cmd/celvet/main.go

```diff
@@ -81,23 +92,58 @@ func main() {
 	}
 
 	limitErrors := celvet.CheckMaxLimits(structural)
+	costErrors, compileErrors, otherErrors := celvet.CheckExprCost(structural)
+	if useJSON {
+		emitJSON(limitErrors, costErrors, compileErrors, otherErrors)
+	} else {
+		emitText(limitErrors, costErrors, compileErrors, otherErrors, *humanReadable)
+	}
+
+	if len(limitErrors)+len(costErrors)+len(compileErrors) > 0 {
+		os.Exit(1)
+	}
+}
+
+func emitJSON(limitErrors []error, costErrors []*celvet.CostError, compileErrors []*celvet.CompilationError, otherErrors []error) {
+	type JSONOutput struct {
+		LimitErrors   []error                    `json:"limitErrors"`
+		CostErrors    []*celvet.CostError        `json:"costErrors"`
+		CompileErrors []*celvet.CompilationError `json:"compileErrors"`
+		OtherErrors   []string                   `json:"otherErrors"`
+	}
+
+	buf := bytes.NewBuffer(nil)
```

Comment:
nit: No need to accumulate the json output in memory before printing.  Just use something like `enc := json.NewEncoder(os.Stdout)` ?

Reply:
Whoops, you're right - fixed